### PR TITLE
test cases and NEWS for #30372 (sparse matmul)

### PR DESCRIPTION
**Pull request by KlausC**
#30372 needs some extra test cases to verify the implementation changes.

### CODE REVIEW

**Comment by ViralBShah** [comment]
I would also mention the performance improvements for sparse matmul in the NEWS.

**Comment by KlausC** [comment]
> Should we also add `a'*b'` for completeness to the tests?

I think, the `as' * bs'` tests the `*(::Adjoint, ::Adjoint)` case already.
I implemented your other hints.

File: NEWS.md [hunk 1]
```diff
@@ -33,11 +33,15 @@ Standard library changes
 
 * Added keyword arguments `rtol`, `atol` to `pinv` and `nullspace` ([#29998]).
 
+#### SparseArrays
+
+* `spmatmul` no longer has the obsolete optional keyword argument `sortindices` ([#30372]).
```
Comment (fredrikekre) [inline NEWS.md]: Isn't this an internal function?

File: NEWS.md [hunk 2]
```diff
@@ -33,11 +33,15 @@ Standard library changes
...
+* performance improvements for `spmatmul` and hence for sparse matrix-matrix multiplication.
```
Comment (fredrikekre) [inline NEWS.md]: AFAICT we usually don't include things like this in news.

File: NEWS.md [hunk 3]
```diff
@@ -33,11 +33,15 @@ Standard library changes
...
+
 #### Dates
 
 * Fixed `repr` such that it displays `DateTime` as it would be entered in Julia ([#30200]).
 
-
```
Comment (fredrikekre) [inline NEWS.md]: Unrelated change.

File: stdlib/SparseArrays/test/sparse.jl [hunk 4]
```diff
@@ -93,10 +93,6 @@ do33 = fill(1.,3)
     end
 end
 
-@testset "Issue #30006" begin
```
Comment (fredrikekre) [inline stdlib/SparseArrays/test/sparse.jl]: Why are you removing this?

File: stdlib/SparseArrays/test/sparse.jl [hunk 5]
```diff
@@ -368,10 +374,6 @@ end
     @test_throws DimensionMismatch dot(sprand(5,5,0.2),sprand(5,6,0.2))
 end
 
-const BASE_TEST_PATH = joinpath(Sys.BINDIR, "..", "share", "julia", "test")
```
Comment (fredrikekre) [inline stdlib/SparseArrays/test/sparse.jl]: Why are you removing the `Quaternion` tests?

File: NEWS.md (hunk 1 above)
Comment (KlausC) [reply NEWS.md]: I inserted this text on demand of the hint:
https://github.com/JuliaLang/julia/pull/30372#issuecomment-447694306

Comment (KlausC) [reply NEWS.md]: will remove references to `spmatmul`.

File: stdlib/SparseArrays/test/sparse.jl (hunk 4 above)
Comment (KlausC) [reply stdlib/SparseArrays/test/sparse.jl]: that was not my intention, sorry - (I did a git fetch upstream before I created the new branch - will be re-inserted.

File: stdlib/SparseArrays/test/sparse.jl (hunk 5 above)
Comment (KlausC) [reply stdlib/SparseArrays/test/sparse.jl]: that was not my intention, sorry - (I did a git fetch upstream before I created the new branch - will be re-inserted.

File: NEWS.md (hunk 2 above)
Comment (KlausC) [reply NEWS.md]: Is a line like this ok or should I leve it out completely?
```
* performance improvements for sparse matrix-matrix multiplication ([#30372]).
```

File: NEWS.md (hunk 1 above)
Comment (ViralBShah) [reply NEWS.md]: @KlausC Sorry to create extra work for you.

File: NEWS.md (hunk 2 above)
Comment (ViralBShah) [reply NEWS.md]: @fredrikekre Why do we not mention performance improvements in NEWS? That generally sounds like users would appreciate it.

Comment (fredrikekre) [reply NEWS.md]: I'm just saying that we have not done it before.

Comment (KlausC) [reply NEWS.md]: Performance improvements can make the difference between usable and unusable in certain application scenarios. What do you recommend?

Comment (ViralBShah) [reply NEWS.md]: @fredrikekre Fair enough. I think it is a good idea and I will merge it now. I also think that performance improvements are generally good to include across the board. If it is something we decide that we shouldn't do, we can always remove it.

**Review by ViralBShah — APPROVED** [review]
